src/Form/Form.js

```diff
@@ -65,8 +71,8 @@ class Form {
         this.activeInput = null
         // We set this to true to skip event listeners while we're autofilling
         this.isAutofilling = false
-        this.handlerExecuted = false
-        this.shouldPromptToStoreData = true
+        this.submitHandlerExecuted = false
+        this.shouldPromptToStoreData = deviceInterface.settings.featureToggles.credentials_saving
```

Comment:
This has defaulted to `true` on all platforms for ages, but still we were not respecting the feature toggle. Fixed now. I've also renamed `handlerExecuted` to make it clearer what it's about.